src/Signature/SignatureV4.php

```diff
@@ -324,11 +324,11 @@ private function getCanonicalizedQuery(array $query)
         ksort($query);
         foreach ($query as $k => $v) {
             if (!is_array($v)) {
-                $qs .= rawurlencode($k) . '=' . rawurlencode($v) . '&';
+                $qs .= rawurlencode($k) . '=' . rawurlencode($v !== null ? $v : '') . '&';
```

Comment:
While this isn't a breaking change, why is it necessary?  `rawurlencode(null)` returns an empty string , same as `rawurlencode('')`.  Did this method change somewhere that I didn't see?  I couldn't find it in the PHP changelogs when I looked, but it's possible I'm not considering something or I missed it.

Reply:
that has difference on PHP8.1 - https://www.php.net/manual/en/migration81.deprecated.php#migration81.deprecated.core.null-not-nullable-internal
https://www.php.net/manual/en/function.rawurlencode.php
So this generates "Deprecation notice" with PHP8.1 and depending on how php&application is configured may lead to unexpected behavior. (i.e application could be configured to halt in case of deprecation notice))

Reply:
Thank you :) that makes sense